### PR TITLE
Added outbox implementation

### PR DESCRIPTION
**Pull request by sawabond**
- Added outbox messages entity
- Added transaction to store user registration and event about user registration to outbox
- Added polling publisher to fetch outbox messages, publish them and mark as published

### CODE REVIEW

File: Auth.PollingPublisher/PollingPublisherService.cs [hunk 1]
```diff
@@ -0,0 +1,56 @@
+﻿using Auth.Infrastructure;
+using Core;
+using Microsoft.EntityFrameworkCore;
+using Newtonsoft.Json;
+
+namespace Auth.PollingPublisher;
+
+public class PollingPublisherService : IHostedService, IDisposable
```
Comment (Puchkof) [inline Auth.PollingPublisher/PollingPublisherService.cs:8]: You can also inherit `BackgroundService`

Comment (Puchkof) [reply Auth.PollingPublisher/PollingPublisherService.cs:8]: I would also consider using Hangfire, popular lib within .NET community

File: Auth.Infrastructure/TransactionFactory.cs [hunk 2]
```diff
@@ -0,0 +1,12 @@
+﻿using Auth.Core.Common;
+using Microsoft.EntityFrameworkCore.Storage;
+
+namespace Auth.Infrastructure;
+
+public class TransactionFactory(AuthDbContext _context) : ITransactionFactory
```
Comment (Puchkof) [inline Auth.Infrastructure/TransactionFactory.cs:6]: The only concern I have that this factory wont work with transient contexts, if someone changes the default lifetime for it

**Review by Puchkof — APPROVED** [review]

File: Auth.Infrastructure/TransactionFactory.cs (hunk 2 above)
Comment (sawabond) [reply Auth.Infrastructure/TransactionFactory.cs:6]: You are correct, thank you. The intention was to create a POC of outbox pattern, so I didn't focus on any other stuff

File: Auth.PollingPublisher/PollingPublisherService.cs (hunk 1 above)
Comment (sawabond) [reply Auth.PollingPublisher/PollingPublisherService.cs:8]: Good idea, for a real implementation scheduled jobs with hangfire would fit great here

**Pull request closed without merging**